app/controllers/reservations_controller.rb

```diff
@@ -1,33 +1,44 @@
 class ReservationsController < ApplicationController
   before_action :set_screening, only: %i[new create]
+  before_action :set_reservation, only: %i[show update]
+
   def new
     @reservation = Reservation.new
-    @screening = Screening.find(params[:screening_id])
   end
 
   def create
     @reservation = Reservation.new(screening_id: params[:screening_id], status: :booked)
 
-    if !params.key?(:seats)
-      @reservation.errors.add(:base, 'Please choose at least one seat')
-      render :new, status: :unprocessable_entity
-
-    else
-      @reservation.save
+    Reservation.transaction do
+      @reservation.save!
       create_tickets
-      redirect_to movies_path
+      redirect_to screening_reservation_path(params[:screening_id], @reservation)
+
+    rescue StandardError
+      render :new, status: :unprocessable_entity
     end
   end
-end
+
+  def update
+    @reservation.update(status: params[:status])
+    redirect_to screening_reservation_path(params[:screening_id], @reservation)
+  end
+
+  def show; end
 
   private
 
-def set_screening
-  @screening = Screening.find(params[:screening_id])
-end
+  def set_screening
+    @screening = Screening.find(params[:screening_id])
+  end
+
+  def set_reservation
+    @reservation = Reservation.find(params[:id])
+  end
 
-def create_tickets
-  params[:seats].each do |seat|
-    Ticket.create(reservation_id: @reservation.id, seat:)
+  def create_tickets
+    params[:seats].each do |seat|
+      Ticket.create(reservation_id: @reservation.id, seat:)
```

Comment:
You forgot to assign anything to seat:, besides we can make this cleaner
```suggestion
      @reservation.tickets.create(seat: seat)
```
You could go one step further and use nested attributes to create reservation with tickets in one query 
https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html

Reply:
Rubocop keeps correcting value assignment, with that output:
`app/controllers/reservations_controller.rb:42:41: C: [Corrected] Style/HashSyntax: Omit the hash value.
      @reservation.tickets.create(seat: seat)`so i left it that way.
https://blog.saeloun.com/2021/09/28/ruby-allow-value-omission-in-hash-literals.html
I'll keep in mind nested attributes suggestion, but I think I'll stick to what I have just for now.